### PR TITLE
Implementing Time Travel Date

### PR DESCRIPTION
**Pull request by szabolcs-keresztes**
## Description and relevant Jira numbers

Adding time travel date to app config and local env.

## Pull Request checklist:

- [ ] [WIP] tag removed from PR title
- [ ] PR has an understandable description

## Git feature branch checklist:

- [ ] branch name comply with our branching strategy
- [ ] git branch contains relevant JIRA ticket number
- [ ] branch rebased against the latest develop
- [ ] commits are squashed

## Sign off process checklist:

- [ ] Code has been tested manually
- [ ] Tests are passing
- [ ] PR link added to JIRA
- [ ] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [ ] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

### CODE REVIEW

File: src/environment/environment.dev.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { EnvironmentFile } from './models/environment.model';
 
 export const environment: EnvironmentFile = {
   isRemote: true,
-  configUrl: 'https://dev.mes.dev-dvsacloud.uk/v1/configuration/dev',
+  configUrl: 'https://szabi.mes.dev-dvsacloud.uk/v1/configuration/dev',
```
Comment (ammarhaiderbjss) [inline src/environment/environment.dev.ts]: needs to be changed back to `dev.`

File: src/pages/journal/journal.reducer.ts [hunk 2]
```diff
@@ -22,15 +21,13 @@ export function journalReducer(state = initialState, action: journalActions.Jour
...
-      ? DateTime.now().format('YYYY-MM-DD')
-      : state.selectedDate;
-
       return {
         ...state,
+
+        // TODO: The reducer has to get the lastRefreshed date from the action
+        // And should not do any logic
         lastRefreshed: (action.onlineOffline ===
           ConnectionStatus.ONLINE  && !action.unAuthenticatedMode) ? new Date() : action.lastRefreshed,
```
Comment (ammarhaiderbjss) [inline src/pages/journal/journal.reducer.ts]: Very true

File: src/environment/environment.dev.ts (hunk 1 above)
Comment (szabolcs-keresztes) [reply src/environment/environment.dev.ts]: just changed it.

File: src/providers/date-time/date-time.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import { DateTime } from '../../shared/helpers/date-time';
+import { AppConfigProvider } from '../app-config/app-config';
+import { isEmpty } from 'lodash';
+
+@Injectable()
+export class DateTimeProvider {
+
+  constructor(private appConfigProvider: AppConfigProvider) {}
+
+  public now(): DateTime {
```
Comment (ammarhaiderbjss) [inline src/providers/date-time/date-time.ts]: Should we also remove the `.now()` method from the date-time helper so we are forced to use this?

**Comment by ammarhaiderbjss** [comment]
@szabolcs-keresztes seems to break `npm run serve:local` :(

File: src/providers/date-time/date-time.ts (hunk 3 above)
Comment (szabolcs-keresztes) [reply src/providers/date-time/date-time.ts]: I see what you mean, I was thinking about that, but the constructor would still be available as public and we need to instantiate DateTime somehow.

What do you think about making the constructor to have required parameter, which would be a `DateTime`, `string` or `Date`?

**Comment by szabolcs-keresztes** [comment]
Fixed the `npm run serve:local`

File: src/providers/date-time/date-time.ts (hunk 3 above)
Comment (rossfauldsbjss) [reply src/providers/date-time/date-time.ts]: could have an optional parameter on the constructor, and default to current date time if not supplied.

Comment (szabolcs-keresztes) [reply src/providers/date-time/date-time.ts]: I am removing `DateTime.now()` which will encourage us to use the `DateTimeProvider.now()` instead.

File: src/pages/journal/components/journal-navigation/__tests__/journal-navigation.spec.ts [hunk 4]
```diff
@@ -53,6 +61,8 @@ describe('JournalNavigationComponent', () => {
                                              ConnectionStatus.ONLINE,
                                              false,
                                              new Date()));
+      console.log('################ date time now', dateTimeProvider.now().format('YYYY-MM-DD'));
```
[the remark this answers is not in the document]
Comment (szabolcs-keresztes) [reply src/pages/journal/components/journal-navigation/__tests__/journal-navigation.spec.ts]: removed it.

**Pull request merged**